certbot/client.py

```diff
@@ -498,6 +498,10 @@ def deploy_certificate(self, domains, privkey_path,
                            "the certificate")
             raise errors.Error("No installer available")
 
+        logger.warning("Future versions of Certbot will automatically configure "
```

Comment:
I personally don't think we should unconditionally print warnings. I think it can create problems with automated alerts if they can't be disabled and it discourages people from paying attention to them.

Can we print this only in the case where we interactively ask the user about a redirect and they say (or default to) no redirect? Alternatively we could do it if we ask at all, but people who want the redirect have nothing to worry about.

This could be done in [enhance_config](https://github.com/certbot/certbot/blob/2812f054a3e9bf87a799c9d153393a9d3593fed0/certbot/client.py#L551) or [redirect_by_default](https://github.com/certbot/certbot/blob/2812f054a3e9bf87a799c9d153393a9d3593fed0/certbot/display/enhancements.py).

We may also want to add simple tests about this behavior.

Reply:
I put it here because we previously talked about users who either have `--quiet` or `--non-interactive` set; are we ok with not notifying these users through Certbot?

Reply:
I may be missing something, but I think my proposal would still work for people with `--quiet` or `--non-interactive` set. If we can't prompt the user, the [IDisplay method in redirect_by_default](https://github.com/certbot/certbot/blob/2812f054a3e9bf87a799c9d153393a9d3593fed0/certbot/display/enhancements.py#L51) should still be called, it will just return the default response which is no redirect.

Reply:
> Can we print this only in the case where we interactively ask the user about a redirect

We don't prompt in `--non-interactive`

Reply:
I see! I should have worded that better. How about:

Can we log this warning only when the user was prompted for a redirect and they said no or they're running non-interactively and they default to no redirect?

This could be by checking for a response from `IDisplay` for no redirect either in [enhance_config](https://github.com/certbot/certbot/blob/2812f054a3e9bf87a799c9d153393a9d3593fed0/certbot/client.py#L551) or [redirect_by_default](https://github.com/certbot/certbot/blob/2812f054a3e9bf87a799c9d153393a9d3593fed0/certbot/display/enhancements.py).

Reply:
> Can we log this warning only when the user was prompted for a redirect and they said no or they're running non-interactively and they default to no redirect?

Sure